Feature/qa 5244 perfezionamento base code

Vincenzo-Massaro opened the pull request:
[QA-5244](https://pagopa.atlassian.net/browse/QA-5244)

[QA-5244]: https://pagopa.atlassian.net/browse/QA-5244?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Vincenzo-Massaro committed:
Various code quality improvements

Changes includes:
-Improved exception handling
-More use of Lombok library where possible
-Improved logging
-Various cleanups and style improvements

Vincenzo-Massaro commented on interop-qa-tests/src/main/java/it/pagopa/interop/agreement/service/impl/AgreementClientImpl.java:
Rimossi attributi di classe inutilizzati

Vincenzo-Massaro commented on interop-qa-tests/src/main/java/it/pagopa/interop/agreement/service/impl/EServiceApiClientImpl.java:
Rimossi attributi di classe inutilizzati

Vincenzo-Massaro commented on interop-qa-tests/src/main/java/it/pagopa/interop/agreement/service/impl/EServiceApiClientImpl.java:
Rimossi attributi di classe inutilizzati

Vincenzo-Massaro commented on interop-qa-tests/src/main/java/it/pagopa/interop/attribute/service/impl/AttributeApiClientImpl.java:
Rimossi attributi di classe inutilizzati

Vincenzo-Massaro commented on interop-qa-tests/src/main/java/it/pagopa/interop/authorization/domain/KeyPairPEM.java:
Classe riformulata sfruttando l'annotazione lombok.Value

Vincenzo-Massaro commented on interop-qa-tests/src/main/java/it/pagopa/interop/authorization/service/factory/SessionTokenFactory.java:
Log migliorata
Usate versioni di metodi riformulati con meno parametri
Scorrimento negli oggetti Map riformulato usando entrySet al posto di keySet
Usate eccezioni dedicate al posto di quelle generiche
Usato costrutto try-with-resources dove necessario
Ritocchi stilistici vari

Vincenzo-Massaro commented on interop-qa-tests/src/main/java/it/pagopa/interop/authorization/service/impl/AuthorizationClientImpl.java:
Rimosso attributo di classe inutilizzato

Vincenzo-Massaro commented on interop-qa-tests/src/main/java/it/pagopa/interop/authorization/service/impl/ProducerClientImpl.java:
Rimossi attributi di classe inutilizzati

Vincenzo-Massaro commented on interop-qa-tests/src/main/java/it/pagopa/interop/authorization/service/utils/CommonUtils.java:
Rimossi attributi di classe inutilizzati
Log migliorata
Ritocchi stilistici vari

Vincenzo-Massaro commented on interop-qa-tests/src/main/java/it/pagopa/interop/authorization/service/utils/KeyPairGeneratorUtil.java:
Formalizzata ulteriormente natura di utility-class attraverso il costruttore privato
Metodi riformulati così da utilizzare solo i parametri realmente utilizzati
Ritocchi vari

Vincenzo-Massaro commented on interop-qa-tests/src/main/java/it/pagopa/interop/config/springconfig/springconfig/InteropRestTemplateConfiguration.java:
Log migliorati
Usato costrutto try-with-resources dove necessario
Ritocchi vari

Vincenzo-Massaro commented on interop-qa-tests/src/main/java/it/pagopa/interop/purpose/RiskAnalysisDataInitializer.java:
Sostituzione di lombok.Getter con lombok.Value
Tolto parametro del costruttore inutilizzato
Usate eccezioni specifiche in sostituzione di quelle generiche

Vincenzo-Massaro commented on interop-qa-tests/src/main/java/it/pagopa/interop/purpose/domain/RiskAnalysis.java:
Classe riformulata usando lombok.Value

Vincenzo-Massaro commented on interop-qa-tests/src/main/java/it/pagopa/interop/purpose/domain/RiskAnalysisDataFromJson.java:
Aggiunte annotazioni lombok ToString e EqualsAndHashCode
Accessibilità del field reimpostato al suo stato originale dopo ogni analisi
Usata eccezione specifica al posto di quella generica

Vincenzo-Massaro commented on interop-qa-tests/src/main/java/it/pagopa/interop/purpose/service/impl/PurposeApiClientImpl.java:
Rimossi attributi di classe inutilizzati

Vincenzo-Massaro commented on interop-qa-tests/src/main/java/it/pagopa/interop/tenant/service/impl/TenantsApiClientImpl.java:
Rimossi attributi di classe inutilizzati

Vincenzo-Massaro commented on interop-qa-tests/src/test/java/it/pagopa/pn/interop/cucumber/steps/DataPreparationService.java:
Riformulata così da usare lombok.AllArgsConstructor
Log migliorata
Ritoccata gestione dei tipi boxed Boolean così da evitare NPEs
Metodi riformulati così da avere solo i parametri effettivamente utilizzati
Ritocchi vari

Vincenzo-Massaro commented on interop-qa-tests/src/test/java/it/pagopa/pn/interop/cucumber/steps/InteropCucumberSpringIntegration.java:
Rimossa KeyPairGeneratorUtil, non essendo un bean ma una plain Java class (possibile si possa applicare lo stesso principio ad altre .class presenti)

mmancini95 commented on interop-qa-tests/src/main/java/it/pagopa/interop/purpose/domain/RiskAnalysisDataFromJson.java:
```diff
@@ -38,26 +39,29 @@ public static class RiskAnalysisAttributes {
             Map<String, List<String>> map = new HashMap<>();
             for (Field field : this.getClass().getDeclaredFields()) {
+                boolean originalAccessibility = field.canAccess(this);
                 field.setAccessible(true);
```
è ancora necessario fare questo set qui visto il nuovo aggiunto sotto?

Vincenzo-Massaro replied:
In teoria questo set serve per memorizzare lo stato di accessibilità del parametro così da poterlo ripristinare sotto, quindi ritengo di sì

mmancini95 commented on interop-qa-tests/src/test/java/it/pagopa/pn/interop/cucumber/steps/InteropCucumberSpringIntegration.java:
```diff
@@ -36,8 +35,7 @@
         ProducerClientImpl.class,
-        KeyPairGeneratorUtil.class,
-        DataPreparationService.class,
+       DataPreparationService.class,
```
wrong indentation

mmancini95 commented on interop-qa-tests/src/test/java/it/pagopa/pn/interop/cucumber/steps/agreement/AgreementCommonSteps.java:
```diff
@@ -51,8 +47,8 @@ public void tenantHasAlreadyCreatedAndPublishedEService(String tenantType, int t
             int randomInt = ThreadLocalRandom.current().nextInt(0, Integer.MAX_VALUE);
-            int TEST_SEED = ThreadLocalRandom.current().nextInt(0, Integer.MAX_VALUE);
-            String eserviceName = String.format("eservice-%d-%d-%d", i, TEST_SEED, randomInt);
+            int testSeed = ThreadLocalRandom.current().nextInt(0, Integer.MAX_VALUE);
+            String eserviceName = String.format("eservice-%d-%d-%d", i, testSeed, randomInt);
```
è possibile rimuovere la generazione di un nuovo valore per testSeed e riutilizzare quello già contenuto in `sharedStepsContext`
```suggestion
            String eserviceName = String.format("eservice-%d-%d-%d", i, sharedStepsContext.getTestSeed(), randomInt);
```

Vincenzo-Massaro committed:
minor fixes

Vincenzo-Massaro committed:
Merge branch 'feature/QA-4772-implementazione-test-capofila' into feature/QA-5244-perfezionamento-base-code

Vincenzo-Massaro committed:
Reorganized CommonUtils.java

Splitted CommonUtils.java to IdentityService.java and PollingService.java: the suffix 'Utils' in Java is traditionally used for static method classes that generally do not participate in dependency injection; since most methods also deal with user identity, a generic 'IdentityService' was chosen, and the makePolling method was moved to the dedicated PollingService.java class
